lib/libutee/tee_api_operations.c

```diff
@@ -1167,10 +1175,11 @@ TEE_Result TEE_MACComputeFinal(TEE_OperationHandle operation,
 		goto out;
 	}
 
-	res = utee_hash_final(operation->state, message, messageLen, mac,
-			      macLen);
+	ml = *macLen;
+	res = utee_hash_final(operation->state, message, messageLen, mac, &ml);
 	if (res != TEE_SUCCESS)
 		goto out;
+	*macLen = ml;
```

Comment:
In case of TEE_ERROR_SHORT_BUFFER, *macLen should be updated (same in other places)

Generally speaking, I think that the output result of the function should have the same behavior as before the patch,  that is having the update of the argument before the error checking.

Then, another patch could optimize this (updating the arguments only when necessary).